### ISSUE TITLE
[JIT][Autocast] Batchnorm folding pass during freezing doesn't preserve types

### ISSUE THREAD

**Issue by davidberard98** [pytorch/pytorch#76578]
### 🐛 Describe the bug

The optimize_numerics pass in the freezing pass interferes with the types used by the autocasting freezing pass.

Autocast() during freezing will cast weights to half precision; but then FoldFrozenConvBatchnorm turns them into full precision, causing errors when the model is run. 

```python
import torch
from typing import Final
net = torch.nn.Sequential(torch.nn.Conv2d(3, 64, (3, 3), 1, 1, bias=False),
                                      torch.nn.BatchNorm2d(64),
                                      torch.nn.PReLU(64)).cuda().eval()
img = torch.randn(3, 3, 112,112).cuda()
with torch.cuda.amp.autocast(True):
    net = torch.jit.trace(net, img, check_trace=False)
    net = torch.jit.freeze(net)
print(net.graph)
feature = net(img)  # forward pass
```

Workaround for now - you can disable optimize_numerics during freezing with `torch.jit.freeze(model, optimize_numerics=False)`

thanks to @xsacha who found the issue & provided the repro

### Versions

master

**Comment by janeyx99** [comment]
TorchScript has been deprecated in favor of torch.export as of 2.11. https://pytorch.org/blog/pytorch-2-11-release-blog/ Closing this issue as we are not further developing on it--please ping if you believe this issue was erroneously closed.
